src/JSInterop/Microsoft.JSInterop.JS/src/src/Microsoft.JSInterop.ts

```diff
@@ -19,42 +19,71 @@ export module DotNet {
   let defaultCallDispatcher: CallDispatcher | null | undefined;
 
   // Provides access to the "current" call dispatcher without having to flow it through nested function calls.
-  let currentCallDispatcher : CallDispatcher | undefined;
+  let currentCallDispatcher: CallDispatcher | undefined;
+
+  /**
+   * Represents a resolved property of a JS object.
+   * 
+   * @param parent The immediate parent object that contains the property.
+   * @param name The name of the property.
+   * @param func Reference to a function if the member is a function. Otherwise, undefined.
+   */
+  interface ObjectMemberDescriptor {
+      parent: any;
+      name: string;
+      func?: Function;
+  }
 
   class JSObject {
-      _cachedFunctions: Map<string, Function>;
+      // TODO(oroztocil): Is it correct to cache functions/properties when they can change?
```

Comment:
Functions could change too, couldn't they?

While it's potentially problematic if something on the path changes, I think we aren't doing anything different here, and this hasn't been a source of problems in the past.